### PR TITLE
Unify http request

### PR DESCRIPTION
**Pull request by wujimin**
unify http request to standard HttpServletRequest, not private interface

### CODE REVIEW

**Comment by WillemJiang** [comment]
@wujimin  Could you take a look the code coverage report? It looks like there are some lines are not covered.

**Comment by wujimin** [comment]
yes, i added so many test case in this PR, but coverage decreased......
i'm handing this problem from this afternoon.

**Comment by wujimin** [comment]
simpler some implements, and delete many useless class in  new  implements, maybe  this  cause the  coverage decreased
from  coverage detail, we can see all coverages for  this  PR file increased

File: demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ClientSignature.java [hunk 1]
```diff
@@ -43,8 +48,12 @@ public Response afterReceiveResponse(Invocation invocation, HttpClientResponse h
     String signature = SignatureUtils.genSignature(responseBodyBuffer);
     String serverSignature = httpResponse.getHeader("signature");
 
-    if (serverSignature != null && !signature.equals(serverSignature)) {
-      return Response.create(Status.UNAUTHORIZED, "signature failed");
+    if (serverSignature != null) {
+      LOGGER.info("check response signature, client: {}, server: {}.", signature, serverSignature);
```
Comment (liubao68) [inline demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ClientSignature.java]: sinature不要记录到日志里面。 属于敏感信息内容

File: demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ServerSignature.java [hunk 2]
```diff
@@ -37,11 +43,17 @@ public int getOrder() {
...
   @Override
   public Response afterReceiveRequest(Invocation invocation, HttpServletRequest request) {
+    if (request instanceof VertxToHttpServletRequest) {
+      return null;
+    }
+
     try {
       String signature = SignatureUtils.genSignature(request.getRequestURI(), invocation.getContext(), request);
       String clientSignature = invocation.getContext("signature");
+      LOGGER.info("check request signature, client: {}, server: {}.", clientSignature, signature);
```
Comment (liubao68) [inline demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ServerSignature.java]: sinature不要记录到日志里面。 属于敏感信息内容

File: demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ClientSignature.java (hunk 1 above)
Comment (wujimin) [reply demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ClientSignature.java]: just  demo, not product

File: demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ServerSignature.java (hunk 2 above)
Comment (wujimin) [reply demo/demo-signature/src/main/java/io/servicecomb/demo/signature/ServerSignature.java]: just  demo, not product

**Pull request merged**